src/kafka_consumer.cpp

```diff
@@ -531,43 +520,40 @@ int KafkaConsumer::execute(int argc, char **argv) {
 
         RdKafka::Message *msg = consumer->consume(1000);
 
-        if (!use_ccb) {
-
-            status = msg_consume(msg, NULL, handler);
-
-            switch (status) {
-                case RdKafka::ERR__TIMED_OUT:
-                    break;
-
-                case RdKafka::ERR_NO_ERROR:
-                    {
-                        const BSM& bsm = handler.get_bsm();
-                        std::stringstream ss;
-                        ss << "Retaining BSM: " << bsm << "\n";
-                        logger->info(ss.str());
-
-                        // if we still have a message in the handler, we send it back out to the producer we have made above.
-                        status = producer->produce(topic, partition, RdKafka::Producer::RK_MSG_COPY, (void *)handler.get_json().c_str(), handler.get_bsm_buffer_size(), NULL, NULL);
-                        if (status != RdKafka::ERR_NO_ERROR) {
-                            logger->error("% Produce failed: " + RdKafka::err2str( status ));
-                        } 
-                    }
-                    break;
-
-                case RdKafka::ERR_INVALID_MSG:
-                    {
-                        const BSM& bsm = handler.get_bsm();
-                        std::stringstream ss;
-                        ss << "Filtering BSM [" << handler.get_result_string() << "] : " << bsm << "\n";
-                        logger->info(ss.str());
-                    }
-                    break;
-
-                default:
-                    ;
-            }
+        status = msg_consume(msg, NULL, handler);
+
+        switch (status) {
+            case RdKafka::ERR__TIMED_OUT:
+                break;
+
+            case RdKafka::ERR_NO_ERROR:
+                {
+                    const BSM& bsm = handler.get_bsm();
+                    std::stringstream ss;
+                    ss << "Retaining BSM: " << bsm << "\n";
+                    logger->info(ss.str());
+
+                    // if we still have a message in the handler, we send it back out to the producer we have made above.
+                    status = producer->produce(topic, partition, RdKafka::Producer::RK_MSG_COPY, (void *)handler.get_json().c_str(), handler.get_bsm_buffer_size(), NULL, NULL);
+                    if (status != RdKafka::ERR_NO_ERROR) {
+                        logger->error("% Produce failed: " + RdKafka::err2str( status ));
+                    } 
+                }
+                break;
+
+            case RdKafka::ERR_INVALID_MSG:
+                {
+                    const BSM& bsm = handler.get_bsm();
+                    std::stringstream ss;
+                    ss << "Filtering BSM [" << handler.get_result_string() << "] : " << bsm << "\n";
+                    logger->info(ss.str());
+                }
+                break;
 
+            default:
+                ;
```

Comment:
Extra semi
```suggestion
                
```

Reply:
This semicolon instructs the default case to do nothing. If we remove it, the program fails to compile.

Since default cases are optional in switch statements, I have removed the default case altogether.